coffee_cmd/src/coffee/mod.rs

```diff
@@ -176,8 +179,20 @@ impl PluginManager for CoffeeManager {
         Err(err)
     }
 
-    async fn list(&mut self) -> Result<(), CoffeeError> {
-        Ok(())
+    async fn list(&mut self, remotes: bool) -> Result<Value, CoffeeError> {
+        let mut plugin_names: Vec<Plugin> = Vec::new();
+        if remotes {
+            for repo in &self.repos {
+                let mut repolist = repo.list().await.unwrap();
+                plugin_names.append(&mut repolist);
+            }
```

Comment:
if we merge this here we lost the track of what is installed and what is not, why if the remote is specified we add a new item in the json result?

such as

```
{
     "plugins": [..],
   "remotes": {
         "local_name": "name of the repository",
         "link": "https....",
          "plugins": []
     },
}
```

this make the command line nicer

Reply:
and also will help when we build the web server

Reply:
Sorry. I don't get it.

So the end goal is that if we run `cargo run -- list` we get also a list of the plugins available to get from the remote repos?

 By the way I noticed this commend in the repository implementation. I think this is relevant?
 ```
    /// list of the plugin installed inside the repository.
    ///
    /// M.B: in the future we want also list all the plugin installed
    /// inside the repository.
    async fn list(&self) -> Result<Vec<Plugin>, CoffeeError> {
        Ok(self.plugins.clone())
    }
 ```

Reply:
this is an old comment, we can remove the `M.B` :)

Reply:
So:

- `cargo run -- list`: list all the plugin installed inside the repository
- `cargo run -- list --remotes`: list all the plugins installed in the repository and all the remotes add to the plugin manager

Reply:
I still can't fully get it. 

I guess this means that the output for `cargo run -- list` stays the same. right?


but for `cargo run -- list --remotes`, the current output is: 
```
{
  "plugins": [
    {
      "conf": null,
      "lang": "Unknown",
      "name": "lightningd",
      "path": "/home/tarekbtc/.coffee/repositories/lightningd",
      "root_path": "/home/tarekbtc/.coffee/repositories/lightningd"
    },
    {
      "conf": null,
      "lang": "Go",
      "name": "sitzprobe",
      "path": "/home/tarekbtc/.coffee/repositories/lightningd/sitzprobe",
      "root_path": "/home/tarekbtc/.coffee/repositories/lightningd/sitzprobe"
    },
    {
      "conf": null,
      "lang": "Go",
      "name": "jwt-factory",
      "path": "/home/tarekbtc/.coffee/repositories/lightningd/jwt-factory",
      "root_path": "/home/tarekbtc/.coffee/repositories/lightningd/jwt-factory"
    }
  ]
}
```
- I just reduced the output to fit here

Can you give me a sample of the required output?

Reply:
```
{
  "plugins": [
    {
      "conf": null,
      "lang": "Unknown",
      "name": "lightningd",
      "path": "/home/tarekbtc/.coffee/repositories/lightningd",
      "root_path": "/home/tarekbtc/.coffee/repositories/lightningd"
    },
    {
      "conf": null,
      "lang": "Go",
      "name": "sitzprobe",
      "path": "/home/tarekbtc/.coffee/repositories/lightningd/sitzprobe",
      "root_path": "/home/tarekbtc/.coffee/repositories/lightningd/sitzprobe"
    },
    {
      "conf": null,
      "lang": "Go",
      "name": "jwt-factory",
      "path": "/home/tarekbtc/.coffee/repositories/lightningd/jwt-factory",
      "root_path": "/home/tarekbtc/.coffee/repositories/lightningd/jwt-factory"
    }
  ],
"remotes" :  {
         "local_name": "name of the repository that you use in remote add",
         "link": "link of the repository",
          "plugins": [] // all the plugins that the repository has
     },
}
```

it is just that!

Reply:
The `cargo run -- list --remotes` for now outputs all plugins in the remote repositories. whether installed or not. Any of these plugins we can install.

Reply:
Yes I understand now. Thanks

Reply:
How can I get the name and the link of the repo? 
I think I can implement getters in `coffee_github/src/repository.rs`

Reply:
the getters need to be implemented inside the coffee_libs/src/repository.rs and then implemented too in coffee_github.

The repository if stored as trait, so you need to have the getters in the trait